components/pattern/src/frontend/mod.rs

```diff
@@ -143,13 +143,13 @@ where
 impl<B: PatternBackend> Pattern<B> {
     #[cfg(feature = "alloc")]
     pub(crate) const fn from_boxed_store_unchecked(store: Box<B::Store>) -> Box<Self> {
-        // Safety: Pattern's layout is the same as B::Store's
+        // Safety: Pattern is repr(transparent) over B::Store
```

Comment:
Do we need to say something about the `PhantomData` or does this speak for itself?

Reply:
I don't think so. repr(transparent) cannot be applied to types where transparency doesn't work.

We do not need to explain the specific semantics of language features, the reviewer should know them, we may need to list the required invariants, and in case of extreme subtlety we might need to say more, but here I don't think there's anything we should say.